### PR TITLE
feat/create-an-endpoint-logic-for-endgame

### PR DESCRIPTION
**Pull request by Alao-Abiodun**
worked on logic to endgame for the endgame route
EndGame

### CODE REVIEW

File: server/src/controllers/game.controller.js [hunk 1]
```diff
@@ -201,8 +201,53 @@ class GameController {
...
+      }
+
+      if (!user_id) {
+        return res
+          .status(400)
+          .send(response("User does not exist", null, false));
+      }
+
+      let winner;
+      if (isGameExist.data.owner) {
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:222]: Your end game logic is wrong…

you are supposed to check if the incoming user_Id is that of owner or opponent. If it is for owner, you set is_owner_win (the one in the model) to true, else you set it to false.

we don’t have a checkmate key in our gameModel or design

Comment (NdubuisiJr) [reply server/src/controllers/game.controller.js:222]: You also need to set the status to ended

File: server/src/controllers/game.controller.js [hunk 2]
```diff
@@ -201,8 +201,53 @@ class GameController {
...
+      let permission;
+
+      const updated = await GameRepo.update(game_id, {
+        ...isGameExist.data,
+        checkmate: {
+          winner,
+        },
+      });
+
+      permission = null;
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:237]: Remove the permission key completely

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]

File: server/src/controllers/game.controller.js (hunk 1 above)
Comment (Alao-Abiodun) [reply server/src/controllers/game.controller.js]: OK @NdubuisiJr

File: server/src/controllers/game.controller.js [hunk 3]
```diff
@@ -201,8 +201,54 @@ class GameController {
...
+      if (user_id === isGameExist.data.owner.user_id) {
+        isGameExist.data.is_owner_winner = true;
+      } else if (user_id == isGameExist.data.opponent.user_id) {
+        isGameExist.data.is_owner_winner = true;
+      }
+
+      // update the Game Info with current result
+      const updated = await GameRepo.update(game_id, {
+        ...isGameExist.data,
+        gameWinner: isGameExist.data.is_owner_winner,
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:236]: we don't have a game winner key. And it's not needed

File: server/src/controllers/game.controller.js [hunk 4]
```diff
@@ -201,8 +201,54 @@ class GameController {
...
+        gameStatus: "ended",
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:237]: it's just `status` not `gameStatus`

File: server/src/controllers/game.controller.js [hunk 5]
```diff
@@ -201,8 +201,54 @@ class GameController {
...
+      });
+
+      const payload = {
+        event: "end_game",
+        winner: updated.gameWinner,
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:242]: winner her should either be `owner object` or `opponent object`

File: server/src/controllers/game.controller.js [hunk 6]
```diff
@@ -201,8 +201,54 @@ class GameController {
...
+        status: updated.gameStatus,
```
Comment (NdubuisiJr) [inline server/src/controllers/game.controller.js:243]: don't forget to change the `gameStatus` to normal `status`

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
once you make these changes we are fine

**Pull request merged**